### PR TITLE
refactoring/new dictionary structure

### PR DESCRIPTION
**Pull request by PedroTroller**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#Contribution guide
+
+1. Run fixers. Your code MUST respect PSR conventions. Please run `make fix` before to submit your contribution.
```
Comment (Shivoham) [inline CONTRIBUTING.md]: in order to submit your contribution

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+2. Run tests. Your code MUST be tested. Please run `make test` before to submit your contribution. If you change a behavior or add new one, please, add your own tests.
```
Comment (Shivoham) [inline CONTRIBUTING.md]: please also run `make test` before submitting your contribution ?

File: src/Knp/DictionaryBundle/Dictionary/StaticDictionaryFactory.php [hunk 3]
```diff
@@ -2,9 +2,10 @@
 
 namespace Knp\DictionaryBundle\Dictionary;
 
+use Knp\DictionaryBundle\Dictionary;
 use Knp\DictionaryBundle\Dictionary\ValueTransformer\TransformerInterface;
 
-final class DictionaryFactory
+final class StaticDictionaryFactory
```
Comment (Shivoham) [inline src/Knp/DictionaryBundle/Dictionary/StaticDictionaryFactory.php]: I'm not sure about this "static dictionary dedicated factory".
As we talked earlier, I think we should keep the DictionaryFactory and introduce dedicated methods.

**Comment by Shivoham** [comment]
@PedroTroller rebase needed :)

File: src/Knp/DictionaryBundle/DependencyInjection/Compiler/DictionaryBuildingPass.php [hunk 4]
```diff
@@ -16,44 +16,37 @@ class DictionaryBuildingPass implements CompilerPassInterface
...
     /**
-     * @param string $class
-     * @param string $class
+     * @param string $name
      * @param array  $dictionary
      *
      * @return Definition
      */
-    private function createDefinition($class, $name, array $dictionary)
+    private function createDefinition($name, array $dictionary)
```
Comment (Shivoham) [inline src/Knp/DictionaryBundle/DependencyInjection/Compiler/DictionaryBuildingPass.php]: Be careful, the Knp\DictionaryBundle\Dictionary\Dictionary use statement is out of date now

Comment (PedroTroller) [inline src/Knp/DictionaryBundle/DependencyInjection/Compiler/DictionaryBuildingPass.php]: Fixed

**Comment by PedroTroller** [comment]
@Shivoham Ready to merge :)

File: spec/Knp/DictionaryBundle/DataFixtures/DictionaryProviderSpec.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    function it_returns_a_key_from_a_dictionary($dictionaries, Dictionary $dictionary)
+    {
+        $dictionaries->get('omg')->willReturn($dictionary);
+        $dictionary->getKeys()->willReturn(array('foo', 'bar', 'baz'));
+
+        $value = $this->dictionary('omg');
+
+        expect(in_array($value->getWrappedObject(), array('foo', 'bar', 'baz')))->toBe(true);
```
Comment (Shivoham) [inline spec/Knp/DictionaryBundle/DataFixtures/DictionaryProviderSpec.php]: `expect(array('foo', 'bar', 'baz'))->toContain($value->getWrappedObject());`

File: spec/Knp/DictionaryBundle/Validator/Constraints/DictionarySpec.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        $this->beConstructedWith(array('name' => 'yolo'));
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Knp\DictionaryBundle\Validator\Constraints\Dictionary');
+    }
+
+    function it_ad_default_values()
```
Comment (Shivoham) [inline spec/Knp/DictionaryBundle/Validator/Constraints/DictionarySpec.php]: it_adds ?

**Comment by Shivoham** [comment]
Huge work @PedroTroller  :beers: !
I've added 2 little comments (not really significants though, it's as you wish).
Otherwise, LGTM, Good to merge, Last night a dj saved my life :heart:

**Comment by PedroTroller** [comment]
:kiss:

**Pull request merged**